packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -165,15 +165,6 @@ export class SummarizerNode implements IRootSummarizerNode {
 		trackState: boolean = true,
 		telemetryContext?: ITelemetryContext,
 	): Promise<ISummarizeResult> {
-		assert(
-			this.isSummaryInProgress(),
-			0x1a1 /* "summarize should not be called when not tracking the summary" */,
-		);
-		assert(
-			this.wipSummaryLogger !== undefined,
-			0x1a2 /* "wipSummaryLogger should have been set in startSummary or ctor" */,
-		);
```

Comment:
curious, why you removed these checks? Looks like some of that moved down, but there is a return statement on (new) line 179, so we will skip them in some cases.

Reply:
This asserts will hit if `summarize` is called directly on `ContainerRuntime` which a lot of tests do. They used to set `trackState: false` which would handle these scenarios. In regular summarization, we still want these conditions to be checked and that will now happen in `completeSummary` step instead of here.

Reply:
So are there cases where we'll do more/too much work because we didn't fail-fast here, but rather wait unitl `completeSummary`?

Reply:
It's possible but there aren't any such cases in prod today.